x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/components/flyout_components/add_to_rules_table/index.tsx

```diff
@@ -80,51 +105,70 @@ const ExceptionsAddToRulesTableComponent: React.FC<ExceptionsAddToRulesComponent
             }
           ),
           multiSelect: 'or' as const,
-          options: rules.flatMap(({ tags }) => {
-            return tags.map((tag) => ({
-              value: tag,
-              name: tag,
-            }));
-          }),
+          options: tagOptions,
         },
       ],
     }),
-    [rules]
+    [tagOptions]
   );
 
-  const sortedRulesBySelection = useMemo(
-    () =>
-      sortBy(rules, [
-        (rule) => {
-          return initiallySelectedRules?.find((initRule) => initRule.id === rule.id);
-        },
-      ]),
-    [initiallySelectedRules, rules]
+  // eslint-disable-next-line react/display-name
+  const LinkRuleSwitch = memo(({ rule }: { rule: Rule }) => {
```

Comment:
should either move this out of the functional ExceptionsAddToRulesTableComponent or just make it a useMemo, as it is now this is a functional component that will get redefined on every render

Reply:
thanks, @kqualters-elastic for the review :) 
I already have the component pushed before in a previous commit but wanted the team to test with me while working on refactoring 😄 https://github.com/elastic/kibana/pull/146121/commits/8bc0c2682fa3b9f36b55c7047c81b7309b7f3be1#diff-a70ea5db87eca5d28626b248345bb344c1527a752fda8504ec21a04bd08169e5R1

Reply:
Please validate the new commit https://github.com/elastic/kibana/pull/146121/commits/cfff10da9aec48ac4f93b321e6de4bbf7cfae65d